homeassistant/components/modbus/base_platform.py

```diff
@@ -230,14 +234,20 @@ def unpack_structure_result(self, registers: list[int]) -> str | None:
                 if isinstance(v_temp, int) and self._precision == 0:
                     v_result.append(str(v_temp))
                 elif v_temp is None:
-                    v_result.append("")  # pragma: no cover
+                    v_result.append("nan")
```

Comment:
Where is the `nan` result handled in the sensor entity?

Reply:
the short answer is "the same way as line 240 is handled, which is not changed in this PR" 😬 

The long answer is "it does not" neither does climate. I will add an extra commit fixing that.

Reply:
And I correct myself. This is not the normal sensor code, this is the part, where the user have requested a comma separated list of multiple values...this is returned as a string in the sensor code.

using "nan" in the string (as already done in line 240) secures that the string returned will look like:
"123,nan,17" and not "123,,17"

Reply:
And how will the sensor parse that string when converting to a sensor state?

Reply:
It will not, it is returned as a string and the user must have some automation that splits it.

Remark, this code is only active if the user configured "structure:" indicating he wants a comma separated string.

For normal sensors thus code is bypassed (see "if len" a couple of lines earlier).

Reply:
It looks like we'll raise a `ValueError` here if we either try to make an integer from `"nan"` or if we try to make a float we'll raise in the base sensor class since we don't allow infinite numbers:

https://github.com/home-assistant/core/blob/aff49cb67abc376d18a58347f13542bfbad81b6c/homeassistant/components/modbus/sensor.py#L124-L130

https://github.com/home-assistant/core/blob/aff49cb67abc376d18a58347f13542bfbad81b6c/homeassistant/components/sensor/__init__.py#L603-L610

Is there some guard for that happening that I'm not seeing?

Reply:
hmmm you got a point, let me change it to a zero.

Reply:
Changed !! 

What I still do not understand, is why the test I made worked

Thanks again for catching this.

Reply:
@MartinHjelmare a polite ping.

Reply:
Regarding the tests, it's the slave sensors that have the coordinator which will use the problematic code path.